sktree/tree/_oblique_splitter.pyx

```diff
@@ -596,3 +597,240 @@ cdef class BestObliqueSplitter(ObliqueSplitter):
+        cdef DTYPE_t[::1]  feature_values = self.feature_values
+        cdef SIZE_t max_features = self.max_features
+        cdef SIZE_t min_samples_leaf = self.min_samples_leaf
+        cdef double min_weight_leaf = self.min_weight_leaf
+
+        # keep track of split record for current_split node and the best_split split
+        # found among the sampled projection vectors
+        cdef ObliqueSplitRecord best_split, current_split
+        cdef double current_proxy_improvement = -INFINITY
+        cdef double best_proxy_improvement = -INFINITY
+
+        cdef SIZE_t p
+        cdef SIZE_t feat_i
+        cdef SIZE_t partition_end
+        cdef DTYPE_t temp_d         # to compute a projection feature value
+        cdef DTYPE_t min_feature_value
+        cdef DTYPE_t max_feature_value
+
+        # Number of features discovered to be constant during the split search
+        cdef SIZE_t n_found_constants = 0
+        cdef SIZE_t n_known_constants = n_constant_features[0]
+        # n_total_constants = n_known_constants + n_found_constants
+        cdef SIZE_t n_total_constants = n_known_constants
+        cdef SIZE_t n_visited_features = 0
+
+        # instantiate the split records
+        _init_split(&best_split, end)
+
+        # Sample the projection matrix
+        self.sample_proj_mat(self.proj_mat_weights, self.proj_mat_indices)
+
+        # For every vector in the projection matrix
+        for feat_i in range(max_features):
+            # Break if already reached max_features
+            if n_visited_features >= max_features:
+                break
+            # Skip features known to be constant
+            if feat_i < n_total_constants:
+                continue
+            # Projection vector has no nonzeros
+            if self.proj_mat_weights[feat_i].empty():
+                continue
+
+            # XXX: 'feature' is not actually used in oblique split records
+            # Just indicates which split was sampled
+            current_split.feature = feat_i
+            current_split.proj_vec_weights = &self.proj_mat_weights[feat_i]
+            current_split.proj_vec_indices = &self.proj_mat_indices[feat_i]
+
+            # Compute linear combination of features and then
+            # sort samples according to the feature values.
+            self.compute_features_over_samples(
+                start,
+                end,
+                samples,
+                feature_values,
+                &self.proj_mat_weights[feat_i],
+                &self.proj_mat_indices[feat_i]
+            )
+
+            # find min, max of the feature_values
+            self.find_min_max(feature_values, &min_feature_value, &max_feature_value)
+
+            # XXX: Add logic to keep track of constant features if they exist
+            if max_feature_value <= min_feature_value + FEATURE_THRESHOLD:
+                n_found_constants += 1
+                n_total_constants += 1
+                continue
```

Comment:
I'm not sure why these lines were added, but this is the reason for the performance regression. All unit-tests do not pass w/ this.

Reply:
I originally thought tracking constant feature was a necessary part of the extratree.. looks like you removed this already.

Reply:
No worries. The constant features is able to be easily tracked in axis-aligned trees, but for oblique trees, this is a bit more difficult, but we would like to have it.

xref: #105